src/me/remie/vulcan/leaguetasks/utils/LeaguePanel.java

```diff
@@ -0,0 +1,119 @@
+package me.remie.vulcan.leaguetasks.utils;
+
+import simple.hooks.simplebot.Game;
+import simple.hooks.wrappers.SimpleWidget;
+import simple.robot.api.ClientContext;
+
+public class LeaguePanel {
+    public ClientContext ctx;
+
+    public LeaguePanel(final ClientContext ctx) {
+        this.ctx = ctx;
+    }
+
+    public boolean doWeNeedToClearFilters() {
+        final SimpleWidget tierDropDownMenu = ctx.widgets.getWidget(657,27);
+        if (!isOptionSelected(tierDropDownMenu, 4, "all")) return true;
+
+        final SimpleWidget typeDropDownMenu = ctx.widgets.getWidget(657,28);
+        if (!isOptionSelected(typeDropDownMenu, 4, "all")) return true;
+
+        final SimpleWidget areaDropDownMenu = ctx.widgets.getWidget(657,29);
+        if (!isOptionSelected(areaDropDownMenu, 4, "all")) return true;
+
+        final SimpleWidget completedDropDownMenu = ctx.widgets.getWidget(657,31);
+        if (!isOptionSelected(completedDropDownMenu, 4, "all")) return true;
```

Comment:
I think we can do the same thing here as I suggested below, we're creating 4 variables and calling a function a maximum of 4 times with the only difference being the child component id. It also seems we use these same child component ids in the `clearAllFilterOptions` function. There's a better way of doing this, maybe we could store some of these ids inside of an enum?